diff --git a/mcproxy/src/proxy/igmp_sender.cpp b/mcproxy/src/proxy/igmp_sender.cpp
index 7f9f745..4d94ae4 100644
--- a/mcproxy/src/proxy/igmp_sender.cpp
+++ b/mcproxy/src/proxy/igmp_sender.cpp
@@ -53,7 +53,14 @@ bool igmp_sender::send_record(unsigned int if_index, mc_filter filter_mode, cons
     if (filter_mode == INCLUDE_MODE && slist.empty() ) {
         m_sock.leave_group(gaddr, if_index);
         return true;
-    } else if (filter_mode == EXLCUDE_MODE || filter_mode == INCLUDE_MODE) {
+    } else if (filter_mode == INCLUDE_MODE) {
+        for (auto & e : slist) {
+            if (! m_sock.join_source_group(gaddr, e.saddr, if_index))
+                return false;
+        }
+
+        return true;
+    } else if (filter_mode == EXCLUDE_MODE) {
         m_sock.join_group(gaddr, if_index);
         std::list<addr_storage> src_list;
         for (auto & e : slist) {
